### PR TITLE
vue-marked instead of vue-markdown?

### PR DESCRIPTION
**Pull request by lab27**
Hi everyone, 

I used this repo as a starting point for a recent project and I loved it. The only problem was the *HUGE* file sizes (like, 500+ kB) on the pages that used VueMarkdown. It has a lot of code!

So last night I tried swapping out *vue-markdown* with *vue-marked* as a mixin and those files are 10x lighter. 

Is there some crucial functionality you want to keep from vue-markdown that vue-marked doesn't have? (I had no need for emojis, for example)

Anyway, take a look and see if something like this could be an option. 

Mad kudos, BTW!

-Nick

ps. This is the first time I'm creating a PR for someone I wasn't in the room with, so I'm not sure what the protocol is. 😊

### CODE REVIEW

**Comment by stefanjudis** [comment]
Hello @lab27. 👋 

This is great and you're absolutely right! I'm all in for getting rid of this huge dependency.

> Is there some crucial functionality you want to keep from vue-markdown that vue-marked doesn't have? (I had no need for emojis, for example)

As long as it works. I'm all in. :)

>  This is the first time I'm creating a PR for someone I wasn't in the room with, so I'm not sure what the protocol is. 

No protocol so far. And I'm happy to help you and ship it. 🎉 

On that note: you opened the PR against a branch which makes the review a little bit difficult. Do you mind opening another PR against master? (I guess this was an accident?)

Thanks a lot for doing this! 👍

**Comment by lab27** [comment]
Sure thing!

File: .contentful.sample.json [hunk 1]
```diff
@@ -1,7 +0,0 @@
-{
```
Comment (stefanjudis) [inline .contentful.sample.json]: Why did you remove this file? 
It's used for onboarding. :)

File: package.json [hunk 2]
```diff
@@ -8,7 +8,9 @@
     "contentful": "^4.3.0",
     "contentful-management": "^3.4.0",
     "nuxt": "^1.0.0-rc4",
-    "vue-markdown": "^2.1.3"
+    "vue-marked": "^0.1.1",
+    "vue-server-renderer": "2.5.11",
+    "vue-template-compiler": "2.5.11"
```
Comment (stefanjudis) [inline package.json]: Maybe I overlook something but are these two (`vue-server-renderer` and `vue-template-compiler`) used right now?

File: pages/blog/_slug.vue [hunk 3]
```diff
@@ -23,8 +23,7 @@
           <time class="tiny">{{ ( new Date(post.fields.publishDate)).toDateString() }}</time>
           <h1>{{ post.fields.title }}</h1>
         </div>
-        <div class="copy">
-          <vue-markdown>{{post.fields.body}}</vue-markdown>
+        <div class="copy" v-html="marked(post.fields.body)">
```
Comment (stefanjudis) [inline pages/blog/_slug.vue]: Personally, I like the component approach better.

```
<div class="copy">
  <markdown>{{post.fields.body}}</markdown>
</div>
```

Because then I don't have to remember `v-html` and `marked`.
What do you think of that?

File: package.json (hunk 2 above)
Comment (lab27) [reply package.json]: Oops. Not sure how those ended up in there. 😊

File: pages/blog/_slug.vue (hunk 3 above)
Comment (lab27) [reply pages/blog/_slug.vue]: I agree with you. It is much more elegant to be able to wrap the post in a component. 

I'm not sure off the top of my head the best way to refactor the `filter` approach into a `component` approach. But I'll happily look into it. Maybe running the filter on the `slot`. If you have any ideas let me know.

Comment (stefanjudis) [reply pages/blog/_slug.vue]: @lab27 Yeah, for me a [single slot](https://vuejs.org/v2/guide/components.html#Single-Slot) with the same logic (`marked(post.fields.body)`) would work for me. :)

**Comment by stefanjudis** [comment]
@lab27 Hey Nick,

are you planning to finish this? Otherwise we could close the PR?
But no pressure, I'd love to have this PR in and am totally fine with keeping it open for a little longer. :)

**Comment by lab27** [comment]
@stefanjudis Hi Stefan,

Yes. I recently got swamped with projects so I got delayed. 

Another hesitation: Looks like `marked` may be too limited in functionality. For example it doesn't seem to `linkify`. Emojis, etc. also. What would you consider the *minimum* feature set of the markdown parser for Contentful Blog? Might need to poke around a bit more, or maybe even roll in some additional libraries to extend `marked`?

n.

**Comment by stefanjudis** [comment]
@lab27 No worries. :)

> For example it doesn't seem to linkify. Emojis, etc. also. What would you consider the minimum feature set of the markdown parser for Contentful Blog? Might need to poke around a bit more, or maybe even roll in some additional libraries to extend marked?

For me this topic would be another PR.

I'm not sure what you mean with `linkify` but `marked` does the job fine with the content model and project implementation today? 

What do you think?

**Comment by stefanjudis** [comment]
@lab27 Hey Nick,

I think we can close this one here or? 

(*I'm cleaning up a bit today.*)

Don't worry about it. Maybe we'll improve it in the future then. :)

Thanks anyways!

**Comment by lab27** [comment]
Hi Stefan,

Yeah, better close it for now. I'll just open a new PR if I figure out how
to integrate Marked as a directive.

Thanks!

Best,
Nick
